c/swig/typedb_driver_csharp.swg

```diff
@@ -0,0 +1,683 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+%csconst(1);
+
+%typemap(cscode) SWIGTYPE %{
+    public $csclassname Released()
+    {
+        var cPtr = swigCPtr.Handle;
+        if (swigCMemOwn)
+        {
+            swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
+            swigCMemOwn = false;
+        }
+
+        return new $csclassname(cPtr, false);
+    }
+
+    public bool IsOwned()
+    {
+        return swigCMemOwn;
+    }
+%}
+
+
+%typemap(csbase) Error "System.Exception";
+%typemap(csbody) Error %{
+    private global::System.Runtime.InteropServices.HandleRef swigCPtr;
+    protected bool swigCMemOwn;
+
+    internal Error(global::System.IntPtr cPtr, bool cMemoryOwn)
+        : base((typedb_driverPINVOKE.error_code(new global::System.Runtime.InteropServices.HandleRef(null, cPtr)) + " "
+            + typedb_driverPINVOKE.error_message(new global::System.Runtime.InteropServices.HandleRef(null, cPtr))).Trim())
+    {
+        swigCMemOwn = cMemoryOwn;
+        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
+    }
+
+    internal static global::System.Runtime.InteropServices.HandleRef getCPtr(Error obj)
+    {
+        return (obj == null)
+            ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero)
+            : obj.swigCPtr;
+    }
+
+    internal static global::System.Runtime.InteropServices.HandleRef swigRelease(Error obj)
+    {
+      if (obj == null)
+      {
+          return new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
+      }
+
+      if (!obj.swigCMemOwn)
+      {
+        throw new global::System.ApplicationException("Cannot release ownership as memory is not owned");
+      }
+
+      global::System.Runtime.InteropServices.HandleRef ptr = obj.swigCPtr;
```

Comment:
It's kinda complicated to import something inside the SWIG file as it's full of layered namespaces. SWIG files tend to be verbose because of it, it is true. I can add `using ...`s locally for methods, but I don't really see a need in it for SWIG files...